src/main/webapp/app/entities/result/result.service.ts

```diff
@@ -104,4 +104,17 @@ export class ResultService {
         }
         return participation;
     }
+
+    /**
+     * This function is used to check whether the student is allowed to submit a complaint or not. Submitting a complaint is allowed within one week after the student received the
+     * result. If the result was submitted after the assessment due date or the assessment due date is not set, the completion date of the result is checked. If the result was
+     * submitted before the assessment due date, the assessment due date is checked, as the student can only see the result after the assessment due date.
+     */
+    isResultOlderThanOneWeek(result: Result, exercise: Exercise): boolean {
```

Comment:
I would suggest naming this similar to the server (isTimeOfComplaintValid) as it does not really check if the result is older than one week but if the complaint can be made

Reply:
I'd also check the same thing - here you do `moment(exercise.assessmentDueDate).isBefore(moment().subtract(1, 'week'));`, while on server `return exercise.getAssessmentDueDate().isAfter(ZonedDateTime.now().minusWeeks(1));` - so you return the opposite thing at the moment (before/after)

Reply:
Well, it checks if the result is older than a week for the student, i.e. if one week has passed since the student actually received the result. But you are right that we should stay consistent. I adjusted the naming of the method and the parameter in the client.